### PR TITLE
WIP cross-compilation support

### PR DESCRIPTION
**Pull request by fasterthanlime**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -43,7 +49,13 @@ semver = "1.0.23"
...
 dialoguer = "0.11.0"
 sha2 = "0.10.6"
-minijinja = { version = "2.3.1", features = ["debug", "loader", "builtins", "json", "custom_syntax"] }
+minijinja = { version = "2.3.1", features = [
+    "debug",
+    "loader",
+    "builtins",
+    "json",
+    "custom_syntax",
+] }
```
[the remark this answers is not in the document]
Comment (fasterthanlime) [reply Cargo.toml:58]: I can revert those and tell zed to Cease Touching TOML for this project

[the remark this answers is not in the document]
Comment (mistydemeo) [reply Cargo.toml]: Hahaha. Yes, please. I've had to turn off a few of the autoformatting defaults as well.

**Comment by fasterthanlime** [comment]
To make this PR thinner, I'd love to get a ✅  on https://github.com/axodotdev/axoasset/pull/212 (and then a cargo publish, and a bump on the version of axoasset that cargo-dist pulls in).

I'd also love to get these merged separately:

  * fast profile: https://github.com/axodotdev/cargo-dist/pull/1470/commits/c249d816f5f082b27b3108f34bc663629487ea52 — this can just be what we make `dev` be, I'd love that actually, then I wouldn't need to build twice. pretty much, I need the deps to be optimized otherwise we're wasting CPU cycles doing xz compression
  * https://github.com/axodotdev/cargo-dist/pull/1470/commits/eb14745dcf61ecb3979aaac881178bf2892f096d — supports a `cargo_dist_url_override` setting, which lets me serve cargo-dist from, say, <https://dl.bearcove.cloud/dump/> in CI — which is how I got [this run](https://github.com/fasterthanlime/bye/actions/runs/11581844804) to finish in 1m48s instead of like, 14m

The rest of the PR is pretty radioactive for now, it has a bunch of .unwrap(), I want to use `Triple` (the structured version, from target_lexicon) more aggressively, but keep the original (non-parsed) version around, just in case we can't parse it?

My guess is that for now if `target_lexicon` cannot parse it, chances are we cannot do anything useful with it anyway, but it just sucks to have to parse it again and again just because the thing we're parsing around is a thinly-veiled string.

**Comment by fasterthanlime** [comment]
Quick piece of insight I had while looking at:

CleanShot 2024-11-01 at 17 05 35@2x

in the absence of other info, we can assume that the custom runners _have the arch they're being used for_.

Since cargo-dist did not support cross when that config format was invented.

Later, we can disallow doing something like:

```toml
[dist.github-custom-runners]
aarch64-unknown-linux-gnu = "my-fast-runner"
x86_64-unknown-linux-gnu = "my-fast-runner"
```

With something like "what's the target triple for github runner 'my-fast-runner'? tell us by doing this:

```toml
[dist.github-custom-runner-specs]
my-fast-runner = { triple = "aarch64-unknown-linux-gnu" }
```

...or whatever makes sense to us.

**Comment by fasterthanlime** [comment]
Closing in favor of #1529

**Pull request closed without merging**